apps/explorer/src/components/common/Button/index.tsx

```diff
@@ -46,205 +25,110 @@ export type ButtonVariations =
-      color: ${white};
-      background: ${mainGradient};
-
-      &:hover {
-        background: ${mainGradientDarker};
-      }
-    `,
-    [Theme.DARK]: css`
-      color: ${white};
-      background: ${mainGradient};
-
-      &:hover {
-        background: ${mainGradientDarker};
-      }
-    `,
-  },
-  get primary() {
-    return this.default
-  },
-  secondary: {
-    [Theme.LIGHT]: css`
-      color: ${blue};
-      background: ${bgLight};
-      border-color: ${blue};
-
-      &:hover {
-        background: ${bgDark};
-      }
-    `,
-    [Theme.DARK]: css`
-      color: ${blue};
-      background: ${bgDark};
-      border-color: ${blue};
-
-      &:hover {
-        color: ${white};
-        background: ${blue};
-      }
-    `,
-  },
-  success: {
-    [Theme.LIGHT]: css`
-      color: ${white};
-      background: ${successLight};
-
-      &:hover {
-        background: ${successDark};
-        border-color: ${successDark};
-      }
-    `,
-    [Theme.DARK]: css`
-      color: ${white};
-      background: ${successDark};
-
-      &:hover {
-        background: ${successLight};
-        border-color: ${successLight};
-      }
-    `,
-  },
-  danger: {
-    [Theme.LIGHT]: css`
-      color: ${white};
-      background: ${dangerLight};
-
-      &:hover {
-        background: ${dangerDark};
-        border-color: ${dangerDark};
-      }
-    `,
-    [Theme.DARK]: css`
-      color: ${white};
-      background: ${dangerDark};
-
-      &:hover {
-        background: ${dangerLight};
-        border-color: ${dangerLight};
-      }
-    `,
-  },
-  warning: {
-    [Theme.LIGHT]: css`
-      color: ${white};
-      background: ${warningLight};
-
-      &:hover {
-        background: ${warningDark};
-        border-color: ${warningDark};
-      }
-    `,
-    [Theme.DARK]: css`
-      color: ${white};
-      background: ${warningDark};
-
-      &:hover {
-        background: ${warningLight};
-        border-color: ${warningLight};
-      }
-    `,
-  },
-  cancel: {
-    [Theme.LIGHT]: css`
-      color: ${white};
-      background: transparent;
-
-      &:hover {
-        background: ${blueDark};
-      }
-    `,
-    [Theme.DARK]: css`
-      color: ${blue};
-      background: transparent;
-
-      &:hover {
-        color: ${whiteDark};
-        background: ${blueDark};
-      }
-    `,
-  },
-  disabled: {
-    [Theme.DARK]: css`
-      color: ${disabledLightOpaque};
-      background: ${disabledLight};
-    `,
-    get light(): ThemeValue {
-      return this[Theme.DARK]
-    },
-  },
-  theme: {
-    [Theme.LIGHT]: css`
-      color: ${white};
-      background: lightsalmon;
-
-      &:hover {
-        color: ghostwhite;
-        background: darkorange;
-      }
-    `,
-    [Theme.DARK]: css`
-      color: ghostwhite;
-      background: purple;
-
-      &:hover {
-        color: ${white};
-        background: darkpurple;
-      }
-    `,
-  },
-})
+const getButtonVariantStyles = (variant: ButtonVariations = 'default') => {
```

Comment:
Not a big deal, just for broadening horizons.
Now the button styles are calculating every time when variant changes.
Would be more efficient to cache the results of `css``` in consts.

Reply:
Addressed now